core/src/main/java/com/spotify/metrics/core/codahale/metrics/ext/MetricRegistryListener.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2016 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.metrics.core.codahale.metrics.ext;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Timer;
+import com.spotify.metrics.core.DerivingMeter;
+
+public interface MetricRegistryListener extends com.codahale.metrics.MetricRegistryListener {
```

Comment:
does this new interface need to exist at all given that `SemanticMetricRegistryListener` exists? I don't see other code in the PR using this new type

Reply:
Yeah, you right we probably don't need so many listeners. However, I added because **SemanticMetricRegistryAdapter** implements MetricRegistryListener instead of SemanticMetricRegistryListener.

Reply:
I did some digging with @ lmuhlha. SemanticMetricRegistryAdapter is used for a Netflix lib that uses Dropwizard registry.  Distribution won't work in that context. I reverted this change and deleted MetricRegistryListener.